src/main/clojure/aerospike_clj/client.clj

```diff
@@ -102,9 +102,9 @@
           ([op-future ce]
            (-> op-future
                (p/then (fn [op-result]
-                         (on-success  ce op-name op-result index op-start-time db))) 
-               (p/catch (fn [op-result]
-                          (on-success ce op-name op-result index op-start-time db))))))]
+                         (on-success  ce op-name op-result    index op-start-time db))) 
+               (p/catch (fn [op-exception]
```

Comment:
you're still calling on-success from p/catch where you should be calling on-failure

Reply:
oops. fixed, once again.